### PR TITLE
Add Contribution Guide

### PR DESCRIPTION
**Pull request by lkiesow**
This patch adds a development and contribution guide, explaining how to
best provide contributions for this project and how contributions will
be handles.

### CODE REVIEW

**Comment by lkiesow** [comment]
Good catch. Any preferences?
I tend to use `sentinel5dl` since that means we can use the same case in commands and documentation.

File: CONTRIBUTING.rst [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+-----------------------------
+
+If you provide a patch, please also provide an explanation of the reasoning behind this patch.
+It is much easier to understand and review code if you know its intention upfront.
+
+If you want to make us happy, please provide this reasoning also as part of your git commit messages.
+
+
+Tests
+-----
```
Comment (shaardie) [inline CONTRIBUTING.rst]: Maybe point to tests directory here.

File: CONTRIBUTING.rst [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+Passing these tests is a requirement for all contributions.
+These tests are also run automatically on our CI system.
+
+If the CI tests on your pull request fail and you are sure it is not caused by your patch, please complain.
+Errors happen and we can easily trigger a new build.
+Your patch cannot be merged without these tests passing.
+
+
+Documentation
+-------------
```
Comment (shaardie) [inline CONTRIBUTING.rst]: Maybe point to `docs` directory here.

File: CONTRIBUTING.rst [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+Being Pragmatic
+---------------
+
+For every rule, there is an exception.
+If you find that there is a good reason one of the following rules does not apply to you,
+please bring it up and explain why.
+We like to be pragmatic if necessary instead of blindly follow rules.
+
+
+Provide Necessary Information
```
Comment (shaardie) [inline CONTRIBUTING.rst]: Issue numbers could be part of the `Provide Necessary Information`

File: CONTRIBUTING.rst [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+Tests
+-----
+
+sentinel5dl comes with a set of unit tests.
+Passing these tests is a requirement for all contributions.
+These tests are also run automatically on our CI system.
+
+To run tests locally, use::
+
+    python -m unittest test
```
[the remark this answers is not in the document]
Comment (lkiesow) [reply CONTRIBUTING.rst]: done.

**Comment by shaardie** [comment]
I am happy thanks.

**Pull request merged**